src/python/pants/backend/python/macros/deprecation_fixers.py

```diff
@@ -0,0 +1,221 @@
+        Get(BuildFileAddress, Address, deps_field.address)
+        for deps_field in python_requirement_dependencies_fields
+    )
+    explicit_deps_per_tgt = await MultiGet(
+        Get(ExplicitlyProvidedDependencies, DependenciesRequest(deps_field))
+        for deps_field in python_requirement_dependencies_fields
+    )
+    deps_per_tgt = await MultiGet(
+        Get(UnexpandedTargets, Addresses(explicit_deps.includes))
+        for explicit_deps in explicit_deps_per_tgt
+    )
+
+    generators = set()
+    generated = {}
+    for python_req_deps_field, build_file_addr, deps in zip(
+        python_requirement_dependencies_fields, build_file_addresses_per_tgt, deps_per_tgt
+    ):
+        generator_tgt = next((tgt for tgt in deps if isinstance(tgt, PythonRequirementsFile)), None)
+        if generator_tgt is None:
+            continue
+
+        # Assume there is a single source. We should have changed this target to use
+        # `SingleSourceField`, alas.
+        generator_source = generator_tgt[MultipleSourcesField].value[0]  # type: ignore[index]
+        if "Pipfile" in generator_source:
+            generator_alias = "pipenv_requirements"
+        elif "pyproject.toml" in generator_source:
+            generator_alias = "poetry_requirements"
+        else:
+            generator_alias = "python_requirements"
+
+        # TODO: Robustly handle if the `name` is already claimed? This can happen, for example,
+        #  if you have two `python_requirements` in the same BUILD file. Perhaps error?
+        generator_name: str | None
+        if (
+            generator_tgt.address.spec_path
+            and generator_tgt.address.spec_path not in dirs_with_default_name
+        ):
+            generator_name = None
+        elif generator_alias == "pipenv_requirements":
+            generator_name = "pipenv"
+        elif generator_alias == "poetry_requirements":
+            generator_name = "poetry"
+        else:
+            generator_name = "reqs"
+
+        generators.add(GeneratorRename(build_file_addr.rel_path, generator_alias, generator_name))
+
+        new_addr = Address(
+            generator_tgt.address.spec_path,
+            target_name=generator_name,
+            generated_name=python_req_deps_field.address.target_name,
+        )
+        generated[python_req_deps_field.address] = (new_addr, generator_alias)
+
+    generators_that_need_renames = sorted(
+        generator for generator in generators if generator.new_name is not None
+    )
+    if generators_that_need_renames:
+        changes = bullet_list(
+            f'`{generator.alias}` in {generator.build_path}: add `name="{generator.new_name}"'
+            for generator in generators_that_need_renames
+        )
+        logger.error(
+            "You must manually add the `name=` field to the following targets. This is not done "
+            f"automatically by the `update-build-files` goal.\n\n{changes}"
+        )
```

Comment:
NB that this rule only ever gets executed once, so this error won't be duplicated. It will also show up above any of the output of the `update-build-files` @goal_rule.